hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/net/TestInetAddressUtils.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.net;
+
+import org.junit.Test;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+
+
+public class TestInetAddressUtils {
+
+  @Test
+  public void testGetCanonicalHostName() throws UnknownHostException {
+    InetAddress localhost = InetAddress.getLocalHost();
+    InetAddress unresolved = InetAddress.getByAddress(localhost.getHostAddress(),
+        localhost.getAddress());
+
+    // Precondition: host name and canonical host name for unresolved returns an IP address.
+    assertEquals(localhost.getHostAddress(), unresolved.getHostName());
```

Comment:
The behavior of this test will vary based on the dns environment at the time it's run. At the very least, you can make this an `assume` statement so that the test will make no assertions when the environmental conditions are inappropriate.

For more rigorous testing, it seems like mocking of a static method becomes possible as of Mockito 3.4.0. Alternatively, I wonder if you can create an environment where you manipulate the property `jdk.net.hosts.file` and provide a file that you populate for the duration of the test.

Reply:
I should specify. At least in OpenJDK11, in the `InetAddress` class, there's a `private static NameService createNameService()` that makes use of this property. I don't know if this is formally documented someplace on the JVM.

Reply:
We're still support Java 8, so I avoided using Java 11 classes.

Reply:
Hmm yes, I see no mention of `jdk.net.hosts.file` in the `InetAddress` of OpenJDK8.